codegen/src/test/resources/software/amazon/awssdk/codegen/poet/rules/endpoint-resolve-interceptor.java

```diff
@@ -147,7 +149,9 @@ private <T extends Identity> SelectedAuthScheme<T> authSchemeWithEndpointSignerP
                     option.putSignerProperty(AwsV4aHttpSigner.DOUBLE_URL_ENCODE, !v4aAuthScheme.disableDoubleEncoding());
                 }
                 if (v4aAuthScheme.signingRegionSet() != null) {
-                    option.putSignerProperty(AwsV4aHttpSigner.REGION_NAME, String.join(",", v4aAuthScheme.signingRegionSet()));
+                    RegionSet regionSet = RegionSet.create(v4aAuthScheme.signingRegionSet().stream()
```

Comment:
I think we could add method to SigV4aAuthScheme that returns RegionSet directly

Reply:
Probably.

Reply:
Yes, I saw 
~~~
List<String> signingRegionSet = sigv4aAuthScheme.signingRegionSet();
RegionSet regionSet = RegionSet.create(signingRegionSet.stream().collect(Collectors.joining(",")));
~~~
and was wondering if there was a reason the auth scheme had to return a list of string.

Reply:
I think we can add this later if we end up using it in more places. Right now, I wouldn't want to add anything to the auth scheme, since it's protected.